datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxDetachNameGenerator.java

```diff
@@ -0,0 +1,33 @@
+package com.sequenceiq.datalake.service.sdx.attach;
+
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.stereotype.Component;
+
+/**
+ * Provides a central logic point for generation of a detached SDX cluster name, as
+ * well as reversing a detached name.
+ */
+@Component
+public class SdxDetachNameGenerator {
+    static final Pattern DETACHED_NAME_PATTERN = Pattern.compile(".*(-[0-9]+)$");
```

Comment:
private missing

Reply:
This pattern is used by the test for this class which is why I didn't make this private.

Reply:
Thank you for the information, I agree with you that basing the accessibility of a field or method off just testing does not fit the reason for seeking privacy in the first place.
I have chosen a solution that is kind of lamer than the ones you offered: I am choosing to just have another copy of the pattern in the test class.

The reason I am doing this is because my approach to testing is not just to ensure things work but also to ensure that in the future, if someone makes changes to the logic, the methods still function in the expected fashion.
So testing `generateNameFromDetached` by using a mock or by calling `generateOriginalNameFromDetached` does not really work for me since that doesn't really verify that the resulting name contains the original name plus an expected string.

Hopefully this makes sense and meets what you think is ideal.